src/Drivers/MiniQMCOptions.cpp

```diff
@@ -0,0 +1,139 @@
+////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source
+// License.  See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 QMCPACK developers.
+//
+// File developed by:
+// Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+//
+// File created by:
+// Peter Doak, doakpw@ornl.gov, Oak Ridge National Lab
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Drivers/MiniQMCOptions.h"
+#include "Utilities/OutputManager.h"
+
+namespace qmcplusplus
+{
+MiniQMCOptions readOptions(int argc, char** argv)
+{
+  MiniQMCOptions mq_opt;
+  using QMCT = QMCTraits;
+  int opt;
+  while (optind < argc)
+  {
+    if ((opt = getopt(argc, argv, "bhjvVa:c:d:g:m:n:N:r:s:t:w:x:z:")) != -1)
+    {
+      switch (opt)
+      {
+      case 'a':
+        mq_opt.splines_per_block = atoi(optarg);
+        break;
+      case 'b':
+        mq_opt.useRef = true;
+        break;
+      case 'c': // number of walkers per crowd
+        mq_opt.crowd_size = atoi(optarg);
+        break;
+      // case 'd': // Device choice
+      //   mq_opt.device_number = atoi(optarg);
+      //   break;
+      case 'g': // tiling1 tiling2 tiling3
+        sscanf(optarg, "%d %d %d", &mq_opt.na, &mq_opt.nb, &mq_opt.nc);
+        break;
+      case 'h':
+        mq_opt.print_help();
+        //should throw
+        mq_opt.valid = false;
+        return mq_opt;
+        break;
+      case 'j':
+        mq_opt.enableJ3 = true;
+        break;
+      case 'm':
+      {
+        const QMCT::RealType meshfactor = atof(optarg);
+        mq_opt.nx *= meshfactor;
+        mq_opt.ny *= meshfactor;
+        mq_opt.nz *= meshfactor;
+      }
+      break;
+      case 'n':
+        mq_opt.nsteps = atoi(optarg);
+        break;
+      case 'N':
+        mq_opt.nsubsteps = atoi(optarg);
+        break;
+      case 'r': // accept
+        mq_opt.accept = atof(optarg);
+        break;
+      case 's':
+        mq_opt.iseed = atoi(optarg);
+        break;
+      case 't':
+        mq_opt.timer_level_name = std::string(optarg);
+        break;
+      case 'v':
+        mq_opt.verbose = true;
+        break;
+      case 'V':
+        ::print_version(true);
+        return mq_opt;
+        //should throw
+        break;
+      case 'w': // number of crowds
```

Comment:
Please keep the meaning of -w unchanged. -w is the number of walkers in the whole population within 1 MPI rank.
The number of crowds is more of an implementation detail and can be derived based on the population size and crowd size.

Reply:
I will leave it and add a different parameter for walkers per crowd.

Reply:
This should work. Could you correct the comment next to case 'w' as well?